src/pages/Search.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { Redirect, useParams, withRouter } from 'react-router-dom';
+import SubredditForm from '../components/SubredditForm';
 import defaultSubreddit from '../helpers/constants';
 
+
 const Search = () => {
-  const { subreddit = defaultSubreddit } = useParams();
-  return (
+  const { subredditQuery } = useParams();
+  const [subreddit, setSubreddit] = useState(subredditQuery || defaultSubreddit);
+  const [route, setRoute] = useState(null);
+
+  const handleSubmit = (evt) => {
```

Comment:
Great that you have consistent naming between the callback handlers. Just a minor suggestion: I'd locate them together similar to this

```
const handleChange = ...

const handleSubmit = ...

useEffect(...)
```